### ISSUE TITLE
parallel-group-params-mismatch caused by browser (reopened)

### ISSUE THREAD

**Issue by craig-dae** [cypress-io/cypress#7199]
Related to https://github.com/cypress-io/github-action/issues/121

First noticed April 30

### Current behavior:

When running tests in parallel on GitHub actions, different containers have different browser versions, causing cypress to fail outright. 

First container: Chrome 80.0.3987.163
Subsequent two containers: 

```
This machine sent the following parameters:

{
  "osName": "linux",
  "osVersion": "Ubuntu Linux - 18.04",
  "browserName": "Chrome",
  "browserVersion": "81.0.4044.122",
```

### Desired behavior:

Tests run in parallel shouldn't have inconsistent configurations

### Test code to reproduce

Job ran as follows:

```
 - name: cypress tests start
        uses: cypress-io/github-action@v1
        with:
          working-directory: ./frontend
          group: ${{ matrix.browser }}-${{ matrix.python-version }}
          browser: ${{ matrix.browser }}
          headless: true
          record: true
          parallel: true
          start: npx ws --spa -s index.html --hostname 0.0.0.0 --port 4200 --directory dist
          wait-on: http://localhost:4200
          wait-on-timeout: 30
        env:
          CYPRESS_RECORD_KEY: ${{ secrets.CYPRESS_RECORD_KEY }}
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

### Versions
Cypress: 4.5.0
Browser: Chrome
GitHub Actions: current

**Comment by craig-dae** [comment]
Also, this seems to happen only in GitHub actions triggered on pushes to a branch, not on PRs. Could be coincidence, but so far, I have not seen it in a PR.

**Comment by jennifer-shehane** [comment]
This is expected behavior in Cypress. We do not run tests across the same group if the browsers are different major versions. Chrome 80 will be considered a different group than Chrome 81 since these are major version differences and could cause a change in how your tests execute. 

You'll need to set up your containers to run on the same major browser version if you want them to be grouped together within Cypress.

**Comment by craig-dae** [comment]
It sounds like you understand GitHub Actions better than me. 

If I have multiple containers set up, shouldn't they all be the same? Why have they suddenly started becoming inconsistent? All the containers are configured identically from my perspective.

**Comment by choonkeat** [comment]
Same

This machine is sending different environment parameters than the first machine that started this parallel run.
...
"browserVersion": "101.0.4951.64"`
whereas the run started by that first machine shows Chrome 102 on https://dashboard.cypress.io/

---

@craig-dae did you overcome this problemz/

**Comment by DmitriiKrupinECM** [comment]
I started getting exactly the same error for Chrome 102 vs 101.0.4951.64 right now in Azure pipelines (we use MS-hosted agents), which is weird because there's the same job configuration that starts both agents in parallel.

**Comment by ckmoga** [comment]
We had these issues for the last two days but seems to have disappeared now.

**Comment by ckmoga** [comment]
For us it turned out that Github had issues: https://www.githubstatus.com/incidents/8xxms3q3qzgp

**Comment by Tofel** [comment]
But that was 6 days ago... I just executed these tests 40 minutes ago :-)

**Comment by Tofel** [comment]
Thanks! I think they did (due to Cypress major version upgrade), but I've forgotten to add that I am not using that official action... (I need to start reverse proxy server before running the tests), so my command looks like this:
```
 bash -c "caddy start --config /root/Caddyfile & sleep 1 && node_modules/cypress/bin/cypress run --browser=chrome --parallel --record --key ${CYPRESS_RECORD_KEY} --spec ${CYPRESS_SPEC_FILE} --group ${CYPRESS_GROUP} --ci-build-id ${CYPRESS_BUILD_ID}"
```

**Comment by ckmoga** [comment]
Nothing looks suspicious for me there;  since the containers were created with a pinned image, I will only suspect Cypress's validation of the parameters is failing. May be start a new issue so they can pick up?

**Comment by Tofel** [comment]
yeah... same image is being used, major version is the same... I think opening a new issue is a good idea, I don't know if they look into closed ones, but before I do that I'll try my luck with Cypress 10.0.02

**Comment by Tofel** [comment]
Okay, I think I know what I did wrong :-) I shouldn't be passing `--parallel` flag, because I do not want Cypress to orchestrate it for me. I just need multiple runs to be grouped together and the orchestration of parallelisation with be done by GH itself via the matrices.
